src/indexEthereum.js

```diff
@@ -194,10 +194,47 @@ export const ethereumCurrencyPluginFactory: EdgeCurrencyPluginFactory = {
           throw new Error('InvalidUriError')
         }
         address = address.replace('/', '') // Remove any slashes
+        let [prefix, contractAddress] = address.split('-') // Split the address to get the prefix according to EIP-681
+        // If contractAddress is null or undefined it means there is no prefix
+        if (!contractAddress) {
+          contractAddress = prefix // Set the contractAddress to be the prefix when the prefix is missing.
+          prefix = 'pay' // The default prefix according to EIP-681 is "pay"
+        }
+        address = contractAddress
         const valid: boolean = EthereumUtil.isValidAddress(address)
         if (!valid) {
           throw new Error('InvalidPublicAddressError')
         }
+        // If the address has a "token-" prefix, it means it's an "Add Token" URI and not a payment one.
+        if (prefix === 'token') {
+          const currencyCode = getParameterByName('symbol', uri)
+          if (!currencyCode || currencyCode.length < 2 || currencyCode.length > 5) {
+            throw new Error('Wrong Token symbol')
```

Comment:
Are you going to need to refactor error messages for internationalization reasons in the future?

Reply:
Yes but that's currently an issue that exists all over in this plugin and in the Bitcoin one too.
So when we do tackle internationalization for the plugins, then we will fix this one as well.